### PR TITLE
Added SwipeToDismissBox for Decompose for Wear OS app

### PR DESCRIPTION
**Pull request by arkivanov**

### CODE REVIEW

File: wearApp/src/main/java/dev/johnoreilly/confetti/wear/navigation/SwipeToDismissBox.kt [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        hasBackground = background != null,
+    ) { isBackground ->
+        val child = background?.takeIf { isBackground } ?: active
+        holder.SaveableStateProvider(child.configuration.key()) {
+            content(child)
+        }
+    }
+}
+
+// Workaround for https://issuetracker.google.com/issues/280392104
```
Comment (yschimke) [inline wearApp/src/main/java/dev/johnoreilly/confetti/wear/navigation/SwipeToDismissBox.kt:83]: This should be fixed in 1.3.0-alpha01 if we want to jump there

https://issuetracker.google.com/issues/280392104

Comment (arkivanov) [reply wearApp/src/main/java/dev/johnoreilly/confetti/wear/navigation/SwipeToDismissBox.kt:83]: Yeah, @joreilly let me know if you are ok to use the alpha version.

Comment (yschimke) [reply wearApp/src/main/java/dev/johnoreilly/confetti/wear/navigation/SwipeToDismissBox.kt:83]: I think we are ok with it.

Comment (arkivanov) [reply wearApp/src/main/java/dev/johnoreilly/confetti/wear/navigation/SwipeToDismissBox.kt:83]: @joreilly could you please review? 😀

File: wearApp/src/main/java/dev/johnoreilly/confetti/wear/navigation/SwipeToDismissBox.kt [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+@Composable
+private fun workaroundBackgroundScrimColor(stackSize: Int): Color {
+    return MaterialTheme.colors.background.copy(alpha = 1F - ((stackSize % 2) * 0.01F))
+}
+
+private fun ChildStack<*, *>.getConfigurations(): Set<String> =
+    items.mapTo(HashSet()) { it.configuration.key() }
+
+private fun Any.key(): String = "${this::class.simpleName}_${hashCode().toString(radix = 36)}"
```
Comment (yschimke) [inline wearApp/src/main/java/dev/johnoreilly/confetti/wear/navigation/SwipeToDismissBox.kt:92]: This looks a bit unexplained but from https://github.com/arkivanov/Decompose/blob/6fa7b41d6f3050f47194d5431eb16a3b4c6e7c10/extensions-compose-jetpack/src/main/java/com/arkivanov/decompose/extensions/compose/jetpack/stack/Children.kt#L57

**Review by yschimke — APPROVED** [review]

File: wearApp/src/main/java/dev/johnoreilly/confetti/wear/navigation/SwipeToDismissBox.kt (hunk 1 above)
Comment (yschimke) [reply wearApp/src/main/java/dev/johnoreilly/confetti/wear/navigation/SwipeToDismissBox.kt:83]: Let's land and follow up, I can bump and remove.

File: wearApp/src/main/java/dev/johnoreilly/confetti/wear/navigation/SwipeToDismissBox.kt (hunk 2 above)
Comment (arkivanov) [reply wearApp/src/main/java/dev/johnoreilly/confetti/wear/navigation/SwipeToDismissBox.kt:92]: Thanks for the review, I will add a comment for this.

**Pull request merged**